### PR TITLE
Whistle signup flow updates

### LINKED ISSUE #1989: Condenser signup process changes for Whistle
We need to make the `Got an account? Login` to trigger `postMessage` if clicked
We need to make the last page of signup process trigger `postMessage` when the last page is activated

Each trigger must give a separate status message/json so that we can differentiate which  `postMessage` is been triggered

### PR DESCRIPTION
**Pull request by relativityboy**
Closes #1989 
Remove navigation that takes users out of signup flow. Remove possibility of 'beta' appearing anywhere.

Of necessity, the UI is very sparse. @pkattera approved.

### CODE REVIEW

File: src/app/Main.js [hunk 1]
```diff
@@ -87,6 +89,7 @@ function runApp(initial_state) {
         window.$STM_csrf = initial_state.offchain.csrf;
         delete initial_state.offchain.csrf;
     }
+    initial_state.app.viewMode = (window.location.search.indexOf(PARAM_VIEW_MODE + '=' + VIEW_MODE_WHISTLE) > -1)? VIEW_MODE_WHISTLE : '';
```
Comment (goldibex) [inline src/app/Main.js]: this ternary expression is really confusing, please rewrite it

File: src/app/components/pages/Approval.jsx [hunk 2]
```diff
@@ -16,6 +17,14 @@ class Approval extends React.Component {
     }
 
     render() {
+        if(process.env.BROWSER && this.props.viewMode === VIEW_MODE_WHISTLE) {
+            try {
+                window.postMessage(WHISTLE_SIGNUP_COMPLETE);
+            } catch(e) {
+                console.error('Cannot window.postMessage when not in whistle app');
```
Comment (goldibex) [inline src/app/components/pages/Approval.jsx]: why does this try/catch block exist?

File: src/app/utils/Links.js [hunk 3]
```diff
@@ -35,8 +35,42 @@ export default {
     ipfsPrefix: /(https?:\/\/.*)?\/ipfs/i,
 }
 
+//todo: possible this should go somewhere else.
```
Comment (goldibex) [inline src/app/utils/Links.js]: 1. please write TODO, not todo
2. please resolve these TODOs. where else would this code go?

File: src/app/utils/Links.test.js [hunk 4]
```diff
@@ -48,6 +48,27 @@ describe('Links', () => {
     })
 })
 
+describe('makeParams', () => {
```
Comment (goldibex) [inline src/app/utils/Links.test.js]: thank you

File: src/server/app_render.jsx [hunk 5]
```diff
@@ -111,8 +111,13 @@ async function appRender(ctx) {
                 offchain.recover_account = account_recovery_record.account_name;
             }
         }
+        const initial_state = {
+            app: {
+                viewMode: (ctx.request.search.indexOf(PARAM_VIEW_MODE + '=' + VIEW_MODE_WHISTLE) > -1)? VIEW_MODE_WHISTLE : ''
```
Comment (goldibex) [inline src/server/app_render.jsx]: refactor ternary expression as requested above

File: src/shared/constants.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export const PARAM_VIEW_MODE = 'view_mode';
```
Comment (goldibex) [inline src/shared/constants.js]: i heartily endorse this methodology for constants.

do not export default an object. this pointlessly removes es6 module magic.

you can really just write
```export const FOO = 'bar';```
it's ok, i promise

File: src/app/components/pages/Approval.jsx (hunk 2 above)
Comment (relativityboy) [reply src/app/components/pages/Approval.jsx]: Because the implementation of postMessage in react-native is different from the spec for web browsers. Calling window.postMessage in a browser with only one argument results in an Error. Because we cannot guarantee a user will never land on this page from a browser, we handle the exception.

File: src/app/utils/Links.js (hunk 3 above)
Comment (relativityboy) [reply src/app/utils/Links.js]: Most likely in a file called utils/Url.js that's currently in the `notifications-ui` branch. Some recent conversations with @bnchdrff suggest that file may be changing or going away.

The TODO is not currently resolvable, but I will capitalize the word for you.

File: src/app/components/pages/Approval.jsx (hunk 2 above)
Comment (goldibex) [reply src/app/components/pages/Approval.jsx]: you're not handling the error, you're ignoring it. why can't you guarantee a user will never land on the page from an error?

File: src/app/utils/Links.js (hunk 3 above)
Comment (goldibex) [reply src/app/utils/Links.js]: if we haven't formally decided to deprecate the file, that's where the code should go.

File: src/app/components/pages/Approval.jsx (hunk 2 above)
Comment (relativityboy) [reply src/app/components/pages/Approval.jsx]: Can you rephrase
> why can't you guarantee a user will never land on the page from an error?
